src/wlapplication.cc

```diff
@@ -1451,164 +1481,159 @@ void WLApplication::handle_commandline_parameters() {
-		}
-		game_type_ = GameType::kReplay;
-		commandline_.erase("replay");
+	// Mutually exclusive options.
+	// These would be better as e.g. "--use_zip=[true|false]", but then we'd have to negate the
+	// boolean value stored in the string, but trueWords and falseWords are hidden in io/profile.cc.
+	// The obvious "--nozip=[true|false]" would be confusing, or at least hard to write a helptext
+	// for.
+	const bool has_nozip = check_commandline_flag("nozip");
+	const bool has_zip = check_commandline_flag("zip");
+	if (has_nozip && has_zip) {
+		throw_exclusive("nozip", "zip");
+	}
+	// Only override config file if we have a command line parameter
+	if (has_nozip || has_zip) {
+		set_config_bool("nozip", has_nozip);
 	}
 
-	if (commandline_.count("new_game_from_template") != 0u) {
-		if (game_type_ != GameType::kNone) {
-		}
-		filename_ = commandline_["new_game_from_template"];
-		if (filename_.empty()) {
-			throw_empty_value("--new_game_from_template");
-		}
-		game_type_ = GameType::kFromTemplate;
-		commandline_.erase("new_game_from_template");
+	// *** End of moved checks ***
+
+	if (check_commandline_flag("verbose")) {
+		g_verbose = true;
 	}
 
-	if (commandline_.count("loadgame") != 0u) {
-		if (game_type_ != GameType::kNone) {
-			throw_exclusive("loadgame");
-		}
-		filename_ = commandline_["loadgame"];
-		if (filename_.empty()) {
-			throw_empty_value("--loadgame");
-		}
-		if (*filename_.rbegin() == '/') {
-			filename_.erase(filename_.size() - 1);
+	const std::map<GameType, std::string> game_type_options = {
+		{ GameType::kEditor, "editor" },
+		{ GameType::kReplay, "replay" },
+		{ GameType::kFromTemplate, "new_game_from_template" },
+		{ GameType::kLoadGame, "loadgame" },
+		{ GameType::kScenario, "scenario" }
+	};
+
+	for (auto pair : game_type_options) {
+		const std::string& opt = pair.second;
+		const bool allow_empty = opt == "editor";
+		auto val = get_commandline_option_value(opt, allow_empty);
+		if (!val.has_value()) {
+			continue;
 		}
-		game_type_ = GameType::kLoadGame;
-		commandline_.erase("loadgame");
-	}
 
-	if (commandline_.count("scenario") != 0u) {
 		if (game_type_ != GameType::kNone) {
-			throw_exclusive("scenario");
-		}
-		filename_ = commandline_["scenario"];
-		if (filename_.empty()) {
-			throw_empty_value("--scenario");
+			throw_exclusive(opt, game_type_options.at(game_type_));
 		}
-		if (*filename_.rbegin() == '/') {
+		game_type_ = pair.first;
+
+		filename_ = *val;
+		if (/* !filename_.empty() && */ *filename_.rbegin() == '/') {
+			// Strip trailing directory separator
 			filename_.erase(filename_.size() - 1);
 		}
-		game_type_ = GameType::kScenario;
-		commandline_.erase("scenario");
 	}
-	if (commandline_.count("script") != 0u) {
-		script_to_run_ = commandline_["script"];
-		if (script_to_run_.empty()) {
-			throw_empty_value("--script");
-		}
+
+	if (auto val = get_commandline_option_value("script"); val.has_value()) {
+		script_to_run_ = *val;
 		if (*script_to_run_.rbegin() == '/') {
```

Comment:
Why `rbegin()` for read-only access? What's wrong with `back()`? :)

Reply:
Oops. :) Though it comes from master actually.